frontend/src/entry/model/settings/ModelRolesInfo.tsx

```diff
@@ -0,0 +1,76 @@
+import { Box, Grid, Typography } from '@mui/material'
+import { Stack } from '@mui/system'
+import { useGetModelRoles } from 'actions/model'
+import { useCallback } from 'react'
+import Loading from 'src/common/Loading'
+import MessageAlert from 'src/MessageAlert'
+import { RoleKind } from 'types/types'
+
+interface ModelRolesInfoProps {
+  modelId: string
+}
+
+export default function ModelRolesInfo({ modelId }: ModelRolesInfoProps) {
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(modelId)
+
+  const roles = useCallback(
```

Comment:
`useCallback` isn't providing any performance gains here. As it stands we'll still be running the internal logic on every render cycle because `roles()` is being called in your JSX block below.

I would `useMemo` for each `kind` and use those stable values in your JSX below. You can also use a reduce so that you don't have to loop over `modelRoles` multiple times. I've made the following generic for entries:
```
  const getFilteredRoles = useCallback(
    (roleKind: string) =>
      entryRoles.reduce<ReactNode[]>((filteredRoles, entryRole) => {
        if (entryRole.kind === roleKind) {
          filteredRoles.push(
            <Box key={entryRole.id}>
              <Typography fontWeight='bold'>{entryRole.name}</Typography>
              <Typography>{entryRole.description}</Typography>
            </Box>,
          )
        }
        return filteredRoles
      }, []),
    [entryRoles],
  )

  const modelRolesList = useMemo(() => getFilteredRoles(RoleKind.MODEL), [getFilteredRoles])
  const schemaRolesList = useMemo(() => getFilteredRoles(RoleKind.SCHEMA), [getFilteredRoles])
```